massa-ledger-worker/src/ledger_db.rs

```diff
@@ -313,7 +316,18 @@ impl LedgerDB {
         db.put_or_update_entry_value(batch, serialized_key, &bytes_bytecode);
 
         // datastore
+        let mut number_of_keys = self
+            .get_datastore_keys(addr, &[])
+            .map(|keys| keys.len())
+            .unwrap_or_else(|| 0);
         for (key, entry) in ledger_entry.datastore {
+            if number_of_keys >= self.max_datastore_entry_count as usize {
+                warn!(
+                    "Too much datastore entries for address {} ({} > {})",
+                    addr, number_of_keys, self.max_datastore_entry_count
+                );
+                continue;
+            }
             if entry.len() > self.max_datastore_value_length as usize {
                 warn!(
```

Comment:
I think that this should be a panic (and below as well)

Reply:
Done.